### DISCUSSION TITLE
Size of NodeId(255), 2 or 4?

### DISCUSSION THREAD

**Discussion by pinterior** [FreeOpcUa/opcua-asyncio#1490] [General]
```
$ python
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from asyncua.ua.uatypes import NodeId
>>> NodeId(255)
NodeId(Identifier=255, NamespaceIndex=0, NodeIdType=<NodeIdType.FourByte: 1>)
>>> NodeId(255).to_binary()
b'\x01\x00\xff\x00'
```

should it be `NodeIdType.TwoByte` and `b'\x00\xff'` ?

(also, `NodeId(0, 255)` should be `NodeIdType.FourByte` instead of `NodeIdType.Numeric` i think)

**Comment by schroeder-** [comment]
The first two bytes signal which nodeidtype is used.

**Reply by pinterior** [reply]
255 is representable in Two Byte NodeId, but if no NodeIdType is specified, NodeIdType.FourByte is automatically selected.
i think both `< 255` below should be `< 256` if the behavior is not intended.

https://github.com/FreeOpcUa/opcua-asyncio/blob/e9978c157d6c68720143211b9d794f2a8cba245e/asyncua/ua/uatypes.py#L456-L464

**Reply by schroeder-** [reply]
Yeah it should be 256
